Map saving parrt 3

Bod9001 opened the pull request:
moves under floor into meta tile map,
improves map saving by reducing number of runtime variables picked up
CL: [Fix] fixes lobby background scaling funny
CL: [Fix] fixes colour picker window being hidden

Bod9001 committed:
Fixes colour  Picker

Bod9001 committed:
Fixeerss lobby screeen background

Bod9001 committed:
Improved maps saaving

Bod9001 committed:
moving  Underfloorffloor iinto Meta tile map

Aranclanos commented on UnityProject/Assets/Scripts/Messages/Server/UpdateTileMessage.cs:
```diff
@@ -45,6 +45,7 @@ public struct delayedData
 
 		public override void Process(NetMessage msg)
 		{
+			if (CustomNetworkManager.IsServer) return;
```
breaks gas overlays I think, unless you took care of it

Bod9001 replied:
ok, will look in to

Bod9001 replied:
looked into seem to be fine, tested with build host to client build ,
noted that you can't see if other people on fire or maybe just host

corp-0 replied:
People on fire should be unrelated. That’s part of the player’s overlays

corp-0 commented on UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:
```diff
@@ -35,7 +36,9 @@ public class SpriteHandler : MonoBehaviour
 
-	[SerializeField, Range(0, 3)]
+	[FormerlySerializedAs("variantIndex"), SerializeField, Range(0, 3)]
+	private int InitialVariantIndex = 0;
```
```suggestion
	private int initialVariantIndex = 0;
```

corp-0 commented on UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/PipeDeconstruction.cs:
```diff
@@ -48,7 +48,7 @@ public override void ServerPerformInteraction(TileApply interaction)
 			for (var i = 0; i < metaDataNode.PipeData.Count; i++)
 			{
-				if (metaDataNode.PipeData[i].RelatedTile != pipeTile) continue;
+				if (metaDataNode.PipeData[i].RelatedTile != pipeTile) continue; //TODO Stuff like layers and stuff can be included
```
what does the TODO mean?

Bod9001 replied:
oh yeah, because pipe layers checking the tile is a same will return incorrect results if they're different layered pipes since they will be still the same tile , just using tile offsets so, it's best to use exact location that will be planned in future as vector4int (xyz for 3D  and then w for the stuff in between layers like underfloor )

Bod9001 committed:
Merge remote-tracking branch 'upstream/develop' into Map-saving-parrt-3

Bod9001 committed:
changes

corp-0 commented on UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:
```diff
@@ -35,7 +36,9 @@ public class SpriteHandler : MonoBehaviour
 	private bool pushTextureOnStartUp = true;
 
-	[SerializeField, Range(0, 3)]
+	[FormerlySerializedAs("InitialVariantIndex"), FormerlySerializedAs("variantIndex"), SerializeField, Range(0, 3)]
```
I imagine you only need one of those FormerlySerializedAs

Bod9001 committed:
changes

Bod9001 committed:
Merge remote-tracking branch 'upstream/develop' into Map-saving-parrt-3

Bod9001 committed:
merge conflicts

corp-0 commented on UnityProject/Assets/Scripts/Objects/Canisters/Canister.cs:
```diff
@@ -145,31 +144,6 @@ private void OnDisable()
-			SetDefaultIntegrity();
-		}
-
-		private void SetDefaultIntegrity()
```
lol what was this?

Bod9001 replied:
idk should never need a automatically setting Armor

Bod9001 committed:
meta tile map fixes

Bod9001 committed:
90% complete

TODO Future, matrix move,  escape shuttle, shuttle  fuel, cargo shuttle , needs to be standard system for more
will work on when doing the loading portion

ThatDan123 commented on UnityProject/Assets/Scripts/MapSaver/MapSaver.cs:
```diff
@@ -417,24 +851,89 @@ public static ObjectMapData SaveObjects(MetaTileMap MetaTileMap)
+										if (mono == null) continue;
+										if (UseInstance)
+										{
+										}
```
This supposed to be empty? Maybe add a TODO

ThatDan123 commented on UnityProject/Assets/Scripts/MapSaver/MapSaver.cs:
```diff
@@ -417,24 +851,89 @@ public static ObjectMapData SaveObjects(MetaTileMap MetaTileMap)
 						continue;
 					}
 
+					//
```
```suggestion
					
```

ThatDan123 commented on UnityProject/Assets/Scripts/MapSaver/MapSaver.cs:
```diff
@@ -471,7 +970,12 @@ public static ObjectMapData SaveObjects(MetaTileMap MetaTileMap)
+					if (MonoSet == null)
+					{
+						Logger.Log("o3o");
+					}
```
Add a TODO to remove if this is needed

ThatDan123 commented on UnityProject/Assets/Scripts/Messages/Server/TileChangesNewClientSync.cs:
```diff
@@ -16,7 +16,7 @@ public struct NetMessage : NetworkMessage
 
 		//just a best guess, try increasing it until the message exceeds mirror's limit
-		private static readonly int MAX_CHANGES_PER_MESSAGE = 20;
+		private static readonly int MAX_CHANGES_PER_MESSAGE = 100;
```
Wasnt this reduced due to length issues?

Bod9001 replied:
looking at the git history was always 10, since the start the offer we think of tile change manager that uses a different message

Bod9001 replied:
./Sprite change message

ThatDan123 replied:
ah ok

ThatDan123 commented on UnityProject/Assets/Scripts/Objects/Doors/GeneralDoorAnimator.cs:
```diff
@@ -2,6 +2,7 @@
 using System.Collections;
 using System.Collections.Generic;
 using TileManagement;
+using Unity.Profiling.LowLevel.Unsafe;
```
?

Bod9001 committed:
Changes

Bod9001 committed:
Forgotten changes
